src/observability/publisher.go

```diff
@@ -1,38 +1,52 @@
 package observability
 
 import (
+	"sync"
+
+	cfg "github.com/accuknox/auto-policy-discovery/src/config"
 	"github.com/accuknox/auto-policy-discovery/src/types"
 )
 
 func ProcessSystemSummary() {
-	if len(PublisherMap) <= 0 {
+
+	if len(SummarizerMap) <= 0 {
 		return
 	}
 
-	PublisherMutex.Lock()
-	// publish summary map in GRPC
-	for ss, sstc := range PublisherMap {
-		var locSummary types.SystemSummary = ss
+	SummarizerMapMutex.Lock()
+	tempSummarizerMap := copyMap(SummarizerMap)
+	SummarizerMapMutex.Unlock()
 
-		// update count/time
-		locSummary.Count = sstc.Count
-		locSummary.UpdatedTime = sstc.UpdatedTime
+	var ProcessSystemSummaryWg sync.WaitGroup
+
+	if cfg.GetCfgObservabilityWriteToDB() {
+		ProcessSystemSummaryWg.Add(1)
+		go UpsertSummaryCronJob(tempSummarizerMap, &ProcessSystemSummaryWg)
+	}
 
-		// publish data to feeder grpc
-		SysSummary.Publish(&locSummary)
+	if cfg.GetCfgPublisherEnable() {
+		PublisherMutex.Lock()
+		// publish summary map in GRPC
+		for ss, sstc := range tempSummarizerMap {
+			var locSummary types.SystemSummary = ss
 
-		// clear each published entry from data map
-		delete(PublisherMap, ss)
+			// update count/time
+			locSummary.Count = sstc.Count
+			locSummary.UpdatedTime = sstc.UpdatedTime
+
+			// publish data to feeder grpc
+			SysSummary.Publish(&locSummary)
+		}
+		PublisherMutex.Unlock()
 	}
-	PublisherMutex.Unlock()
+	ProcessSystemSummaryWg.Wait()
 }
 
-func updatePublisherMap() {
-	for ss, sstc := range SummarizerMap {
-		PublisherMap[ss] = types.SysSummaryTimeCount{
-			Count:       PublisherMap[ss].Count + sstc.Count,
-			UpdatedTime: sstc.UpdatedTime,
-		}
-		delete(SummarizerMap, ss)
+func copyMap[K, V comparable](m map[K]V) map[K]V {
```

Comment:
```suggestion
func moveMap[K, V comparable](m map[K]V) map[K]V {
```
Move this function under `common` pkg.